src/ray/raylet/task.cc

```diff
@@ -0,0 +1,55 @@
+#ifndef TASK_CC
+#define TASK_CC
+
+#include "task.h"
+
+#include "common.h"
+
+namespace ray {
+
+const TaskExecutionSpecification &Task::GetTaskExecutionSpec() const {
+  return task_execution_spec_;
+}
+
+const TaskSpecification &Task::GetTaskSpecification() const {
+  return task_spec_;
+}
+
+const std::vector<ObjectID> Task::GetDependencies() const {
+  std::vector<ObjectID> dependencies;
+  for (int i = 0; i < task_spec_.NumArgs(); ++i) {
+    int count = task_spec_.ArgIdCount(i);
+    for (int j = 0; j < count; j++) {
+      dependencies.push_back(task_spec_.ArgId(i, j));
+    }
+  }
+  auto execution_dependencies = task_execution_spec_.ExecutionDependencies();
+  dependencies.insert(dependencies.end(), execution_dependencies.begin(), execution_dependencies.end());
+  return dependencies;
+}
+
+bool Task::DependsOn(ObjectID object_id) const {
+  // Iterate through the task arguments to see if it contains object_id.
+  int64_t num_args = task_spec_.NumArgs();
+  for (int i = 0; i < num_args; ++i) {
```

Comment:
i++

Reply:
this is insignificant here, but, generally, prefix increment is preferred in C++ for performance reasons. Quoting:
https://stackoverflow.com/questions/3181211/prefix-postfix-increment-operators
"It is typical that a postfix operator will be worse on performance because you have to create another copy before doing the increment (and this is why I've gotten in the habit of always using prefix unless I need something else)."

Reply:
Yes, I always do ++i and move on to life for the exact reason @atumanov mentioned.

Reply:
I see, I'm aware of the performance argument, but I thought people preferred `i++` for aesthetic reasons and because compilers take care of the performance issue. Anyway, I'm fine with `++i`.